console/src/components/Services/Data/TablePermissions/Permissions.js

```diff
@@ -89,6 +91,19 @@ class Permissions extends Component {
     dispatch(fetchFunctionInit());
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` will be deprecated and in React 16.3 there was `getDerivedStateFromProps` introduced to replace it. So can we use this instead?
```js
static getDerivedStateFromProps(nextProps, prevState) {
    const currPermissionsState = prevState.permissionsState;
    const nextPermissionsState = nextProps.permissionsState;

    if (
      currPermissionsState.role !== nextPermissionsState.role ||
      currPermissionsState.query !== nextPermissionsState.query
    ) {
      return { filterString: '' };
    }
    return null;
}
```

Reply:
I didnt go down that route as I would then have to store the previousProps in state which I wanted to avoid. Is there any other way I am not aware of?

In any case, we should be converting all these to hooks anyway when we refactor so it might not be worth adding a lot of code to achieve this

Reply:
Figured out an easy workaround